src/Sulu/Bundle/MediaBundle/Resources/public/js/components/area-selection/main.js

```diff
@@ -0,0 +1,487 @@
+/*
+ * This file is part of the Sulu CMS.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+/**
+ * Component for selecting areas (for example for cropping an image)
+ *
+ * @class AreaSelection
+ * @constructor
+ */
+define(['underscore', 'jquery', 'text!./frame.html'], function(_, $, frameTemplate) {
+
+    'use strict';
+
+    var defaults = {
+        eventNamespace: 'sulu.area-selection',
+        instanceName: '',
+        image: null,
+        areaGuidingWidth: null,
+        areaGuidingHeight: null,
+        data: null
+    },
+
+    translations = {
+        minimumSizeReached: 'sulu-media.minimum-size-reached'
+    },
+
+    /**
+     * Raised when the component has been successfully initialized.
+     * @event sulu.area-selection.initialized
+     */
+    INITIALIZED = function() {
+        return createEventName.call(this, 'initialized');
+    },
+
+    /**
+     * Raised when the area the component selects has changed.
+     * @event sulu.area-selection.area-changed
+     */
+    AREA_CHANGED = function() {
+        return createEventName.call(this, 'area-changed');
+    },
+
+    /**
+     * Listens on and sets the area-guide dimensions of the component
+     *
+     * @param {Number} areaGuidingWidth The new area-guide-width
+     * @param {Number} areaGuidingHeight The new area-guide-height
+     * @param {Object} data The new data to use
+     *
+     * @event sulu.area-selection.change-area-guide-dimensions
+     */
+    SET_AREA_GUIDE_DIMENSIONS = function() {
+        return createEventName.call(this, 'set-area-guide-dimensions');
+    },
+
+    /**
+     * returns normalized event names
+     */
+    createEventName = function(postFix) {
+        return this.options.eventNamespace +
+            '.' + (!!this.options.instanceName ? this.options.instanceName + '.' : '') + postFix;
+    };
+
+    return {
+
+        initialize: function() {
+            this.options = this.sandbox.util.extend(true, {}, defaults, this.options);
+
+            this.$frame = null;
+            this.$backdrop = null;
+
+            // The original width and height of the image.
+            this.originalWidth = null;
+            this.originalHeight = null;
+            this.areaGuidingWidth = this.options.areaGuidingWidth;
+            this.areaGuidingHeight = this.options.areaGuidingHeight;
+            this.physicalareaGuidingWidth = null;
+            this.physicalareaGuidingHeight = null;
```

Comment:
should probably be `physicalAreaGuiding...`, and what kind of value does it exactly hold?